# Type: fix
# Subtype: Ix
# English Summary: Fix the bug in BaseApiProvider, do not add ordinary setter/getter as "instance methods"
# Chinese Summary: 修复BaseApiProvider的bug，不要将普通的setter/getter添加为“实例方法”
# Entry Number: 1

# === Git Diff Content ===

diff --git a/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/metaclass/BaseApiProvider.java b/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/metaclass/BaseApiProvider.java
index 84d74a4d88..b8b7c3f01c 100644
--- a/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/metaclass/BaseApiProvider.java
+++ b/grails-core/src/main/groovy/org/codehaus/groovy/grails/commons/metaclass/BaseApiProvider.java
@@ -24,6 +24,7 @@ import java.util.Arrays;
 import java.util.List;
 
 import org.apache.commons.lang.ArrayUtils;
+import org.codehaus.groovy.grails.commons.GrailsClassUtils;
 import org.codehaus.groovy.reflection.CachedClass;
 import org.codehaus.groovy.reflection.CachedMethod;
 import org.codehaus.groovy.runtime.metaclass.ReflectionMetaMethod;
@@ -112,6 +113,12 @@ public abstract class BaseApiProvider {
         if (EXCLUDED_METHODS.contains(name)) return false;
 
         boolean isStatic = Modifier.isStatic(modifiers);
+        
+        // skip plain setters/getters by default for instance methods (non-static)
+        if(!isStatic && (GrailsClassUtils.isSetter(name, method.getParameterTypes()) || GrailsClassUtils.isGetter(name, method.getParameterTypes()))) {
+            return false;
+        }        
+        
         int minParameters = isStatic ? 0 : 1;
 
         return Modifier.isPublic(modifiers) &&
